.github/scripts/rust/lint.rs

```diff
@@ -0,0 +1,281 @@
+
+    if !contents.contains(PREFIX) || !contents.contains(SUFFIX) {
+        panic!("malformed .cargo/config.toml, please add the required markers")
+    }
+
+    let file = contents.lines()
+        .map(|value| value.trim())
+        .skip_while(|value| *value != PREFIX)
+        .take_while(|value| *value != SUFFIX)
+        .filter(|value| !value.is_empty())
+        .filter(|value| value.starts_with('"'))
+        // remove leading `"` and trailing `",`
+        .map(|value| &value[1..(value.len() - 2)]).fold(LintFile::default(), |mut acc, line| {
+            let (arg, lint) = line.split_at(2);
+
+            match arg {
+                "-A" => acc.allow.insert(lint.to_owned()),
+                "-W" => acc.warn.insert(lint.to_owned()),
+                "--force-warn" => acc.force_warn.insert(lint.to_owned()),
+                "-D" => acc.deny.insert(lint.to_owned()),
+                "-F" => acc.forbid.insert(lint.to_owned()),
+                _ => panic!("unrecognized rust flag {arg}")
+            };
+
+            acc
+        });
+
+    file
+}
+
+fn generate(cwd: &Path) {
+    let path = cwd.join(".cargo/config.toml");
+
+    if !path.exists() || !path.is_file() {
+        panic!(".cargo/config.toml needs to exist")
+    }
+
+    let lints = collect_lints(cwd);
+
+    // read the `config.toml`
+    let contents = fs::read_to_string(&path).expect(&format!("should be able to read {}", path.display()));
+
+    if !contents.contains(PREFIX) || !contents.contains(SUFFIX) {
+        panic!(concat!(
+            "malformed .cargo/config.toml, please add the required markers ",
+            r###""## START CLIPPY LINTS ##" and "## END CLIPPY LINTS ##" on a separate line to "###,
+            "delimit the region this script is allowed to modify."
+        ))
+    }
+
+    // strip out content between the markers
+    let prefix = contents.lines().take_while(|value| value.trim() != PREFIX);
+    let suffix = contents.lines().skip_while(|value| value.trim() != SUFFIX);
+
+    let indent = contents.lines()
+        .find(|value| value.trim() == PREFIX)
+        .map(|value| " ".repeat(value.len() - value.trim_start().len()))
+        .unwrap_or_else(|| "    ".to_owned());
+
+    let body = lints.forbid.into_iter().map(|lint| format!(r#"{indent}"-F{lint}","#))
+        .chain(lints.deny.into_iter().map(|lint| format!(r#"{indent}"-D{lint}","#)))
+        .chain(lints.force_warn.into_iter().map(|lint| format!(r#"{indent}"--force-warn{lint}","#)))
+        .chain(lints.warn.into_iter().map(|lint| format!(r#"{indent}"-W{lint}","#)))
+        .chain(lints.allow.into_iter().map(|lint| format!(r#"{indent}"-A{lint}","#)));
+
+    let contents: Vec<_> = prefix.map(|line| line.to_owned())
+        .chain(once(format!("{indent}## START CLIPPY LINTS ##")))
+        .chain(body)
+        .chain(suffix.map(|line| line.to_owned()))
+        .collect();
+
+    fs::write(&path, contents.join("\n")).expect("unable to write `.cargo/config.toml`");
```

Comment:
```suggestion
    fs::write(&path, format!("{}\n", contents.join("\n"))).expect("unable to write `.cargo/config.toml`");
```

Reply:
fixed differently in [6a91206](https://github.com/hashintel/hash/pull/2075/commits/6a91206d27365ba06e2103d2dc9a916703ebd562)